nemo/collections/asr/parts/utils/eval_utils.py

```diff
@@ -0,0 +1,387 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import json
+import os
+import subprocess
+from pathlib import Path
+from typing import Tuple
+
+from omegaconf import DictConfig, OmegaConf, open_dict
+
+from nemo.collections.asr.metrics.wer import word_error_rate_detail
+from nemo.utils import logging
+
+
+def run_asr_inference(cfg: DictConfig) -> DictConfig:
+    if (cfg.model_path and cfg.pretrained_name) or (not cfg.model_path and not cfg.pretrained_name):
+        raise ValueError("Please specify either cfg.model_path or cfg.pretrained_name!")
+
+    if cfg.inference_mode.mode == "offline":
+        cfg = run_offline_inference(cfg)
+
+    elif cfg.inference_mode.mode == "chunked":
+        if (
+            "total_buffer_in_secs" not in cfg.inference_mode
+            or "chunk_len_in_secs" not in cfg.inference_mode
+            or not cfg.inference_mode.total_buffer_in_secs
+            or not cfg.inference_mode.chunk_len_in_secs
+        ):
+            raise ValueError(
+                f"Please specify both total_buffer_in_secs and chunk_len_in_secs for chunked inference_mode"
+            )
+        cfg = run_chunked_inference(cfg)
+
+    elif cfg.inference_mode.mode == "offline_by_chunked":
```

Comment:
What's the fundamental difference between "chunked" and "offline_by_chunked"? It seems that the only difference is only that "offline_by_chunked" will set default parameters while "chunked" will not. Could you please add more docstrings to explain the different use cases?

Reply:
what you set is totally correct. Added more explanation in docstring.